ofrak_ai/ofrak_ai/chatgpt.py

```diff
@@ -0,0 +1,31 @@
+import os
+
+from dataclasses import dataclass
+from typing import Optional
+
+from ofrak.model.component_model import ComponentConfig
+from ofrak.model.resource_model import ResourceAttributes
+
+
+@dataclass
+class ChatGPTConfig(ComponentConfig):
+    """
+    :param api_key: the OpenAI API key to use
+    :param api_organization: the OpenAI API organization to use
+    :param model: the OpenAI model to use
+    :param system_message: a message which can be prepended to the conversation in API calls, which
+        is used for providing additional information to ChatGPT for generating responses
+    :param temperature: a measure of the randomness of ChatGPT's responses, between 0 (low) and
+        2 (high)
+    """
+
+    api_key: Optional[str] = os.getenv("OPENAI_API_KEY")
+    api_organization: Optional[str] = os.getenv("OPENAI_ORGANIZATION")
+    model: str = "gpt-3.5-turbo"
```

Comment:
I haven't played around with changing the model much, but it would certainly change the cost of the query and likely the speed as well. From what I've read, it could also affect things like how well ChatGPT follows constraints. For instance, GPT-4 is reportedly better at following constraints than 3.5. I think testing with different models would be worthwhile. One consequence of that would be: how do we want to present the results of that testing to potential users of this modifier? Presumably they will be making the choice of which model to use based on their own cost-benefit analysis.

Reply:
I think making this an enum instead of a string (with the same default value as now), and including a docstring that briefly mentions some trade-offs. Probably doesn't need to be too extensive.

Part of what I'm thinking about is that in the context of [this PR](https://github.com/redballoonsecurity/ofrak/pull/287), it would be easier for a user to fill out this config if they could select a model option from a drop-down instead of needing to know which magic string to put in.

Reply:
Makes total sense, I'll do that